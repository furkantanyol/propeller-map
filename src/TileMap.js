import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { setZoomLevel, getColsAndRowsNumber } from './helpers/map'

export default class TileMap extends Component {
  static propTypes = {
    zoom: PropTypes.number.isRequired
  }

  state = {
    zoomLevel: setZoomLevel(this.props.zoom) || 0,
    cols: getColsAndRowsNumber(this.zoomLevel),
    rows: getColsAndRowsNumber(this.zoomLevel)
  }

  getTileSrc = (col, row) => {
    const { zoomLevel } = this.state
    const tileSrc = `./assets/tiled/${zoomLevel}/${col}/${row}.jpg`
    return tileSrc
  }

  getTiles = () => {
    const { rows, cols, zoomLevel } = this.state
    let tiles = []
    for (let row = 0; row < rows; row++) {
      for (let col = 0; col < cols; col++) {
        let tileSrc = this.getTileSrc(col, row)
        tiles.push(<Tile key={tileSrc} src={tileSrc} zoomLevel={zoomLevel} />)
      }
    }

    return tiles
  }

  // updateTiles(zoomLevel, cols, rows) {
  //   if (this.zoomLevel === zoomLevel) {
  //   }
  // }

  render() {
    const { zoomLevel } = this.state
    const tiles = this.getTiles()
    debugger
    return (
      <div className='map-container' id='map-container'>
        <div
          className={`tiles-wrapper ${zoomLevel === 3 ? 'zoom-level-3' : ''}`}
        >
          {tiles}
        </div>
      </div>
    )
  }
}

const Tile = ({ key, src, zoomLevel }) => {
  debugger
  return (
    <img key={key} src={src} className={`tile-zl-${zoomLevel}`} alt='tile' />
  )
}
