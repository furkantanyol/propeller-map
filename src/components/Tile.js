import React from 'react'
import PropTypes from 'prop-types'
import { TILE_LENGTH } from '../util/constants'

export default function Tile({ src, colIndex, rowIndex }) {
  return (
    <img
      style={{
        position: 'absolute',
        left: `${colIndex > 0 ? colIndex * TILE_LENGTH : 0}px`,
        top: `${rowIndex > 0 ? rowIndex * TILE_LENGTH : 0}px`
      }}
      src={src}
      alt='tile'
    />
  )
}

Tile.propTypes = {
  src: PropTypes.string.isRequired,
  colIndex: PropTypes.number.isRequired,
  rowIndex: PropTypes.number.isRequired
}
