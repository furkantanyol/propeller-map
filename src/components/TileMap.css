.tiles-wrapper {
  position: relative;
  transition: 0.395s all cubic-bezier(0, 0, 0.2, 1);
}

.tiles {
  visibility: hidden;
  opacity: 0;
  height: 100%;
  transition: 0.395s all cubic-bezier(0, 0, 0.2, 1);
}

.tiles-active {
  opacity: 1;
  visibility: visible;
}
