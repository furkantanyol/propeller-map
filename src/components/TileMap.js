import React from 'react'
import PropTypes from 'prop-types'
import { getContainerSize } from '../util/helpers'
import './TileMap.css'
import Loading from './Loading'
import { TILE_LENGTH } from '../util/constants'

export default function TileMap({ zoom, zoomLevel, tiles, isLoading }) {
  const children = []
  if (tiles && !isLoading) {
    // Create tiles wrappers when Map Type tiles is loaded and handle zoomLevel change
    tiles.forEach((value, key) => {
      children.push(
        <div
          key={key}
          className={`tiles ${zoomLevel - 1 === key ? 'tiles-active' : ''}`}
        >
          {value}
        </div>
      )
    })
  }

  if (isLoading || !tiles) return <Loading />

  return (
    <div
      style={{
        width: `${getContainerSize(TILE_LENGTH, zoomLevel - 1)}px`,
        height: `${getContainerSize(TILE_LENGTH, zoomLevel - 1)}px`,
        transform: `scale(${zoom})`
      }}
      className='tiles-wrapper'
    >
      {children}
    </div>
  )
}

TileMap.propTypes = {
  zoom: PropTypes.number.isRequired,
  zoomLevel: PropTypes.number.isRequired
}
