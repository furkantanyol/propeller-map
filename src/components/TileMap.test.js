import React from 'react'
import Tile from './Tile'
import TileMap from './TileMap'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tiles = new Map()

  tiles.set(
    1,
    <Tile src={'https://via.placeholder.com/256'} colIndex={1} rowIndex={1} />
  )
  tiles.set(
    2,
    <Tile src={'https://via.placeholder.com/256'} colIndex={1} rowIndex={1} />
  )

  const tree = renderer
    .create(<TileMap zoom={1} zoomLevel={1} tiles={tiles} isLoading={false} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
