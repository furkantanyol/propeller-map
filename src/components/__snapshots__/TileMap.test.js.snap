// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="tiles-wrapper"
  style={
    Object {
      "height": "256px",
      "transform": "scale(1)",
      "width": "256px",
    }
  }
>
  <div
    className="tiles "
  >
    <img
      alt="tile"
      src="https://via.placeholder.com/256"
      style={
        Object {
          "left": "256px",
          "position": "absolute",
          "top": "256px",
        }
      }
    />
  </div>
  <div
    className="tiles "
  >
    <img
      alt="tile"
      src="https://via.placeholder.com/256"
      style={
        Object {
          "left": "256px",
          "position": "absolute",
          "top": "256px",
        }
      }
    />
  </div>
</div>
`;
